libvast/src/system/pivoter.cpp

```diff
@@ -0,0 +1,159 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/system/pivoter.hpp"
+
+#include "vast/command.hpp"
+#include "vast/concept/parseable/to.hpp"
+#include "vast/concept/parseable/vast/expression.hpp"
+#include "vast/detail/string.hpp"
+#include "vast/expression.hpp"
+#include "vast/logger.hpp"
+#include "vast/system/exporter.hpp"
+#include "vast/table_slice.hpp"
+
+#include <caf/atom.hpp>
+#include <caf/event_based_actor.hpp>
+
+namespace vast::system {
+
+namespace {
+
+caf::optional<record_field>
+common_field(const pivoter_state& st, const record_type& indicator) {
+  auto f = st.cache.find(indicator);
+  if (f != st.cache.end())
+    return f->second;
+    // TODO: This algorithm can be enabled once we have a live updated
+    //       type registry. (Switch the type of target to record_type.)
+#if 0
+  for (auto& t : target.fields) {
+    for (auto& i : indicator.fields) {
+      if (t.name == i.name) {
+        st.cache.insert({indicator, i});
+        return i;
+      }
+    }
+  }
+#else
```

Comment:
I would preface the following with a clear comment that this is an interim heuristic and nothing that stays.